support-frontend/assets/helpers/redux/checkout/product/state.ts

```diff
@@ -0,0 +1,75 @@
+import type {
+	ContributionType,
+	OtherAmounts,
+	SelectedAmounts,
+} from 'helpers/contributions';
+import { detect as detectCountryGroup } from 'helpers/internationalisation/countryGroup';
+import type { IsoCurrency } from 'helpers/internationalisation/currency';
+import type { BillingPeriod } from 'helpers/productPrice/billingPeriods';
+import type { FulfilmentOptions } from 'helpers/productPrice/fulfilmentOptions';
+import type { ProductOptions } from 'helpers/productPrice/productOptions';
+import type {
+	ProductPrice,
+	ProductPrices,
+} from 'helpers/productPrice/productPrices';
+import { getCurrency } from 'helpers/productPrice/productPrices';
+import type { SubscriptionProduct } from 'helpers/productPrice/subscriptions';
+
+export type GuardianProduct =
```

Comment:
This is a bit of a fudge for now- in the future it would be good to actually define the specific product types in this file (or in another single place where other product types are defined) and have these types follow a more coherent pattern.